lib/trento/support/ecto/payload.ex

```diff
@@ -1,27 +1,60 @@
 defmodule Trento.Support.Ecto.Payload do
   @moduledoc """
   Ecto Type that represents a JSONB payload that contains an array or a map.
+
+  Opts:
+  - keys_as_atoms (boolean): Converts map keys to existing atoms
   """
 
-  use Ecto.Type
-  def type, do: :map
+  alias Trento.Support.StructHelper
+
+  use Ecto.ParameterizedType
+
+  def type(_params), do: :map
+
+  def init(opts) do
+    keys_as_atoms = Keyword.get(opts, :keys_as_atoms, false)
+
+    validate_opts(keys_as_atoms)
+
+    %{
+      keys_as_atoms: keys_as_atoms
+    }
+  end
 
   # Provide custom casting rules.
-  def cast(data) when is_list(data) or is_map(data) do
+  def cast(data, _params) when is_list(data) or is_map(data) do
     {:ok, data}
   end
 
+  def cast(nil, _), do: {:ok, nil}
```

Comment:
This `nil` matches returning `{:ok, nil}` are required as we might store `null` entries in the database as well.
Besides that, the previous `Ecto.Type` was already bypassing the `nil` value, so this copies the behaviour.